test/cppsuite/test_harness/workload_generator.h

```diff
@@ -43,120 +44,38 @@ namespace test_harness {
  */
 class workload_generator : public component {
     public:
-    workload_generator(configuration *configuration, timestamp_manager *timestamp_manager,
-      workload_tracking *tracking)
-        : component(configuration), _timestamp_manager(timestamp_manager), _tracking(tracking)
+    workload_generator(configuration *configuration, database_operation *db_operation,
+      timestamp_manager *timestamp_manager, workload_tracking *tracking)
+        : component(configuration), _database_operation(db_operation),
+          _timestamp_manager(timestamp_manager), _tracking(tracking)
     {
     }
 
     ~workload_generator()
     {
         for (auto &it : _workers)
             delete it;
+        delete _database_operation;
```

Comment:
The operation is passed by the test in a number of scenarios couldn't this unintentionally delete our users data?

Reply:
As discussed, this should be deleted where it is created, i.e at the test level